godtools/App/Features/AppLanguage/Data-DomainInterface/GetDownloadableLanguagesListRepository.swift

```diff
@@ -0,0 +1,100 @@
+//
+//  GetDownloadableLanguagesListRepository.swift
+//  godtools
+//
+//  Created by Rachael Skeath on 12/15/23.
+//  Copyright © 2023 Cru. All rights reserved.
+//
+
+import Foundation
+import Combine
+
+class GetDownloadableLanguagesListRepository: GetDownloadableLanguagesListRepositoryInterface {
+    
+    private let languagesRepository: LanguagesRepository
+    private let downloadedLanguagesRepository: DownloadedLanguagesRepository
+    private let getTranslatedLanguageName: GetTranslatedLanguageName
+    private let resourcesRepository: ResourcesRepository
+    private let localizationServices: LocalizationServices
+    
+    init(languagesRepository: LanguagesRepository, downloadedLanguagesRepository: DownloadedLanguagesRepository, getTranslatedLanguageName: GetTranslatedLanguageName, resourcesRepository: ResourcesRepository, localizationServices: LocalizationServices) {
+        
+        self.languagesRepository = languagesRepository
+        self.downloadedLanguagesRepository = downloadedLanguagesRepository
+        self.getTranslatedLanguageName = getTranslatedLanguageName
+        self.resourcesRepository = resourcesRepository
+        self.localizationServices = localizationServices
+    }
+    
+    func getDownloadableLanguagesPublisher(currentAppLanguage: AppLanguageDomainModel) -> AnyPublisher<[DownloadableLanguageListItemDomainModel], Never> {
+        
+        return Publishers.CombineLatest(
+            languagesRepository.getLanguagesChanged(),
+            downloadedLanguagesRepository.getDownloadedLanguagesChangedPublisher()
+        )
+        .map { _ in
+            
+            return self.languagesRepository.getLanguages().map { language in
+                                
+                let languageNameInOwnLanguage = self.getTranslatedLanguageName.getLanguageName(
+                    language: language,
+                    translatedInLanguage: language.id
+                )
+                let languageNameInAppLanguage = self.getTranslatedLanguageName.getLanguageName(
+                    language: language,
+                    translatedInLanguage: currentAppLanguage
+                )
+                
+                let toolsAvailableText = self.getToolsAvailableText(for: language.languageCode, translatedIn: currentAppLanguage)
+                
+                let downloadStatus = self.getDownloadStatus(for: language.id)
+                
+                return DownloadableLanguageListItemDomainModel(
+                    languageId: language.id,
+                    languageNameInOwnLanguage: languageNameInOwnLanguage,
+                    languageNameInAppLanguage: languageNameInAppLanguage,
+                    toolsAvailableText: toolsAvailableText,
+                    downloadStatus: downloadStatus
+                )
+            }
```

Comment:
Hey @rachaelblue also see about sorting this list, I included the notes from Figma in the Jira ticket.  It seems they want sorting by download order first and then remaining non downloaded languages sorted alphabetically.  I'm not sure which language though?  I'd guess app language, but may want to confirm with Jill on the sorting.

Reply:
Confirmed with Jill to sort by app language translation.